### PR TITLE
Server/Client basic TCP communication

### PR DESCRIPTION
**Pull request by m-kuzmin**
# Changelog
## Server
- Binding to port and accepting connections
- Reading client messages
- Responding to clients
- On <kbd>Ctrl</kbd> + <kbd>C</kbd> closes all connections and stops reading any messages

## Client
- Connecting to server
- Sending messages
- Receiving responces
- In case of an error closes the connection (considers an error to always be the server closing the connection)

### CODE REVIEW

File: cmd/client.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package main
```
Comment (penkova) [inline cmd/client.go]: you should have another structure for the project:

cmd:
--client:
----main.go
--server:
----main.go

File: cmd/server.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		address = "localhost"
+		port    = "8080"
+	)
+
+	ln, err := net.Listen("tcp", address+":"+port)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Println("Server ready for connections, address:", ln.Addr())
+	defer ln.Close()
```
Comment (penkova) [inline cmd/server.go]: how about using channels for graceful shutdown?

Comment (m-kuzmin) [reply cmd/server.go]: I dont think its applicable here since the Listener should stop when main returns.

**Comment by m-kuzmin** [comment]
Dont merge yet please, I'll add additional graceful shutdown logic to the server.

File: cmd/server.go (hunk 2 above)
Comment (m-kuzmin) [reply cmd/server.go]: I think this is resolved now? (8a56c5e)

File: cmd/client.go (hunk 1 above)
Comment (m-kuzmin) [reply cmd/client.go]: Resolved in https://github.com/m-kuzmin/golang-tcp-client-and-server/pull/1/commits/4717acc9a8c2f3911616dd67ee3f5773f20e3dcb

**Comment by m-kuzmin** [comment]
If everything is resolved we can merge now

File: cmd/server/main.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+		wg.Add(1)
+		go handleConnection(conn, &wg, quitCh)
+	}
+
+	wg.Wait()
+}
+
+func handleConnection(conn net.Conn, wg *sync.WaitGroup, quitCh <-chan struct{}) {
+	address := conn.RemoteAddr()
```
Comment (penkova) [inline cmd/server/main.go:59]: why do you need connection close here?

**Review by penkova — APPROVED** [review]

**Pull request merged**